### DISCUSSION TITLE
Is it possible to reduce performance consumption of DoUpdate function in MeshDrawPacket.cpp

### DISCUSSION THREAD

**Discussion by RogerShaw** [o3de/o3de#17385] [Help Wanted]
My scene consists of hundreds of shaderballs, all using the same material instance. I found DoUpdate function in MeshDrawPacket.cpp comsumes a lot of cpu resources. This causes FPS drops (60fps->30~50fps)
Is it possible to reduce performance consumption of DoUpdate function in MeshDrawPacket.cpp when we change properties of material instance used in meshes?
Thanks so much!

**Comment by VickyAtAZ** [comment]
Have you tried to enable mesh instancing?
You can check this wiki page for details: https://github.com/o3de/o3de/wiki/%5BAtom%5D-Mesh-Instancing:-For-Content-Creators

**Reply by RogerShaw** [reply]
I am trying to do this test. Thank you!
